modules/tensor_mechanics/src/materials/cohesive_zone_model/BiLinearMixedModeTraction.C

```diff
@@ -0,0 +1,212 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "BiLinearMixedModeTraction.h"
+#include "libmesh/utility.h"
+
+registerMooseObject("TensorMechanicsApp", BiLinearMixedModeTraction);
+
+InputParameters
+BiLinearMixedModeTraction::validParams()
+{
+  InputParameters params = CZMComputeLocalTractionTotalBase::validParams();
+  params.addRequiredParam<Real>("penalty_stiffness", "Penalty stiffness.");
+  params.addRequiredParam<Real>("GI_C", "Critical energy release rate in normal direction.");
+  params.addRequiredParam<Real>("GII_C", "Critical energy release rate in shear direction.");
+  params.addParam<Real>("viscosity", 0.0, "Viscosity.");
+  params.addRequiredParam<Real>("normal_strength", "Tensile strength in normal direction.");
+  params.addRequiredParam<Real>("shear_strength", "Tensile strength in shear direction.");
+  params.addRequiredParam<Real>("eta", "The B-K power law parameter.");
+  params.addCoupledVar("normal_strength_scale_factor", "Scale factor for normal strength.");
+  params.addClassDescription("Mixed mode bilinear traction separation law.");
+  return params;
+}
+
+BiLinearMixedModeTraction::BiLinearMixedModeTraction(const InputParameters & parameters)
+  : CZMComputeLocalTractionTotalBase(parameters),
+    _scale_factor(isCoupled("normal_strength_scale_factor")
```

Comment:
why not just default `normal_strength_scale_factor` to 1?

Reply:
done.